spring.go

```diff
@@ -0,0 +1,405 @@
+package beego
+
+import (
+	"errors"
+	"github.com/astaxie/beego/context"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+const (
+	TagName = "autowired"
+
+	PreHandling  = "preHandling"
+	PostHandling = "postHandling"
+
+	InterfaceType      = "InterfaceType"
+	ImplementationType = "ImplementationType"
+	BeanName           = "BeanName"
+
+	PanicOnError   = "panicOnError"
+	NilOnError     = "nilOnError"
+	SaveErrorToCtx = "saveErrorToCtx"
+)
+
+var (
+	TypeMismatch         = errors.New("type mismatch")
```

Comment:
warning: exported var TypeMismatch should have comment or be unexported (golint)